eng/resolveContract.targets

```diff
@@ -60,11 +61,38 @@
                                       ReferenceAssembly="" />
     </ItemGroup>
   </Target>
-  
+
+  <!-- Generate a .txt file with all public types of the project referenced by ProjectForGenAPIDocIdGeneration (e.g. System.Private.CoreLib or System.Private.Uri implementation assembly).
+       This file is then later on passed to GenAPI as the list of types which should be "split out" with the conditional compilation logic. -->
+  <Target Name="GenerateConditionalTypeList"
+          Condition="'@(ProjectReference->WithMetadataValue('Identity', '$(ProjectForGenAPIDocIdGeneration)'))' != '' and ('$(ContractTypesPartiallyMovedToCoreLib)' == 'true' or '$(ContractTypesPartiallyMovedToUri)' == 'true')">
+    <PropertyGroup>
+      <!-- generate conditional compilation symbols to hide types implemented in contracts so we can include the .cs in the System.Private.CoreLib ref -->
+      <GenAPIConditionalTypeList>$(IntermediateOutputPath)GenAPIConditionalTypeList.txt</GenAPIConditionalTypeList>
+      <GenAPIConditionalTypeListSymbol Condition="'$(GenAPIConditionalTypeListSymbol)' == ''">!HIDE_CONTRACT_TYPES</GenAPIConditionalTypeListSymbol>
+      <GenAPIConditionalTypeListWrapOtherTypes Condition="'$(GenAPIConditionalTypeListWrapOtherTypes)' == ''">true</GenAPIConditionalTypeListWrapOtherTypes>
+    </PropertyGroup>
+    <ItemGroup>
+      <_genAPIResolvedCoreLibProjectReferenceDir Include="@(_ResolvedProjectReferencePaths->WithMetadataValue('MSBuildSourceProjectFile','$(CoreLibProject)')->'%(RootDir)%(Directory)')" />
```

Comment:
This won't return anything if `ProjectForGenAPIDocIdGeneration` is not CoreLib and when CoreLib isn't referenced.

Reply:
that's true, but that can't really happen since the target only runs when `$(ContractTypesPartiallyMovedToCoreLib)' == 'true' or '$(ContractTypesPartiallyMovedToUri)' == 'true'` (the first property implies you have a CoreLib reference and the second one only applies to Uri which also has one).

Reply:
Right, I understand that this works as-is today but it is fragile. Maybe you can come up with a better pattern to retrieve the corelib assembly or include an explicit error that checks that corelib is referenced and suggests to reference it if it isn't.

Reply:
`@(ReferencePath->'%(RootDir)%(Directory)'->Distinct())` works as well (that's what the GenAPI targets from arcade does), so will switch to that